Adds exploration point api

jeppe-style opened the pull request:
as per title

First implementation for https://github.com/benchflow/benchflow/issues/418

depends on PR https://github.com/benchflow/benchflow/pull/512

TODO 
- [x] if there is no exploration space there should not be any reference to the exploration space
- [x] getting exploration space returns 404 when deployed

jeppe-style committed:
Fixes code style issues

jeppe-style commented:
if there is no exploration space there should not be any reference to the exploration space

jeppe-style committed:
Fixes issue that the exploration space configuration link was also shown when no exploration space

jeppe-style committed:
Fixes issue that ExplorationPointResource did not work properly

jeppe-style committed:
Code style fixes

VincenzoFerme commented on benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/models/explorationspace/MongoCompatibleExplorationSpace.java:
```diff
@@ -94,4 +96,29 @@ public JavaCompatExplorationSpace toJavaCompat() {
+    ExplorationSpacePointResponse explorationSpacePointResponse =
+        new ExplorationSpacePointResponse();
+
+    if (index >= size || index < 0) {
```
This should be an error, not an empty response.

jeppe-style replied:
I don't see how this changes the functionality

VincenzoFerme replied:
It becomes a proper API, with proper response codes.

VincenzoFerme replied:
Postponed in https://github.com/benchflow/benchflow/issues/494

VincenzoFerme commented on benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/models/explorationspace/MongoCompatibleExplorationSpace.java:
```diff
@@ -94,4 +96,29 @@ public JavaCompatExplorationSpace toJavaCompat() {
+      return explorationSpacePointResponse;
+    }
+
+    usersDimension
```
Why is the following not exposed as part of the APIs of the DSL?

jeppe-style replied:
I don't get it. Why should the response be part of the DSL?

VincenzoFerme replied:
My fault

VincenzoFerme commented on benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/resources/BenchFlowTestResource.java:
```diff
@@ -212,7 +221,39 @@ public BenchFlowTestModel getBenchFlowTestStatus(@PathParam("username") String u
+
+        String url;
+
+        if (explorationSpace == null || explorationSpace.getSize() == 0) {
```
The field should not be visualised in the response in this case

jeppe-style replied:
I suggest to move this into an issue since it is not important for the use cases.

VincenzoFerme replied:
Postponed in https://github.com/benchflow/benchflow/issues/494

VincenzoFerme committed:
Adds resource for exploration point
Adds link to exploration point configuration when getting test status

# Conflicts:
#	benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/models/explorationspace/MongoCompatibleExplorationSpace.java

VincenzoFerme committed:
Merge remote-tracking branch 'origin/devel' into feature-tm-exploration-point-api

# Conflicts:
#	benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/BenchFlowTestManagerApplication.java
#	benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/models/explorationspace/MongoCompatibleExplorationSpace.java
#	benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/BenchFlowTestManagerApplicationIT.java
#	benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/resources/BenchFlowTestResourceTest.java
#	benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/scheduler/TestTaskSchedulerIT.java
#	benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/services/internal/dao/UserDAOIT.java

VincenzoFerme committed:
Fixes Merge Diffs and Code Cleanup
